src/app/pages/form/form.component.ts

```diff
@@ -1,9 +1,72 @@
 import { Component } from '@angular/core';
+import { FormBuilder, Validators } from '@angular/forms';
+
+interface City {
+    name: string;
+  //   code: string;
+}
 
 @Component({
-    selector: 'app-form',
-    templateUrl: './form.component.html'
+  selector: 'app-form',
+  templateUrl: './form.component.html',
+  styleUrls: ['./form.component.scss'],
 })
 export class FormComponent {
+  userForm = this.fb.group({
+    location: ['', Validators.required],
+    destination: [''],
+    selectedCity: [''],
+    departureDate: [''],
+    returnDate: [''],
+    passenger: [''],
+
+  });
+  citiesFrom: any[] | undefined;
```

Comment:
`citiesFrom` oraz `citiesTo` zamieniamy na 
```
cities: City[] = [
//tutaj wszystkie obiekty miast
]

```

Nie musisz w onInicie wypełniać tablic danymi, możesz zrobić to już w tym miejscu :)

Reply:
i wystrzegaj się jak ognia używania `any`. Używamy tego w baaaardzo wyjątkowych sytuacjach, zawsze używamy typów